mycroft/configuration/config.py

```diff
@@ -265,8 +283,20 @@ def load_config_stack(configs=None, cache=False, remote=True):
 
         # Merge all configs into one
         base = {}
-        for c in configs:
-            merge_dict(base, c)
+        for cfg in configs:
+            # check for protected keys in remote config (changes blocked by system)
+            if isinstance(cfg, RemoteConf):
+                if skip_remote:  # remote config disabled at system level
+                    continue
+                # delete protected keys from remote config
+                flattened_delete(cfg, protected_remote)
+            # check for protected keys in user config (changes blocked by system)
+            elif isinstance(cfg, LocalConf) and cfg.path in xdg_locations + [OLD_USER_CONFIG]:
+                if skip_user:  # user config disabled at system level
+                    continue
+                # delete protected keys from user config
+                flattened_delete(cfg, protected_user)
+            merge_dict(base, cfg)
```

Comment:
this is beautiful

should it be a function? it would pre-implement additional config layers, though i have no suggestions as to what those might be

Reply:
i can move it into a function if that feels cleaner, this method is already quite big anyway

Reply:
actually, moving this into a func is ugly, it needs `get_xdg_config_locations` and `_get_system_constraints` so those would need to be called twice or passed in args, regardless it would no longer be beautiful :P

Reply:
It would certainly need some params, but I don't see those

The interesting part would be creating a clean reference to `skip_n` and `protected_n` for like values of `n`